### PR TITLE
TornadoMath double version of trigonometric functions added

### PR DESCRIPTION
**Pull request by jjfumero**
#### Description

Some of the `FP64` version of the trigonometric functions within the `TornadoMath` class were missing.  Related issue #481 

#### Problem description

n/ a,.

#### Backend/s tested

Mark the backends affected by this PR.

- [X] OpenCL
- [X] PTX
- [X] SPIRV

#### OS tested

Mark the OS where this PR is tested.

- [X] Linux
- [ ] OSx
- [ ] Windows

#### Did you check on FPGAs?

If it is applicable, check your changes on FPGAs.

- [ ] Yes
- [X] No

#### How to test the new patch?

```bash
tornado-test -V -pk --fast uk.ac.manchester.tornado.unittests.math.TestTornadoMathCollection
```

### CODE REVIEW

**Comment by jjfumero** [comment]
On NVIDIA 1050;


```
Test: class uk.ac.manchester.tornado.unittests.math.TestTornadoMathCollection
	Running test: testTornadoMathSignumDouble ................  [PASS] 
	Running test: testTornadoMathSignumDoubleNaN ................  [OPENCL CONFIGURATION UNSUPPORTED] 
	Running test: testTornadoMathSignumFloatNaN ................  [OPENCL CONFIGURATION UNSUPPORTED] 
	Running test: testTornadoMathCos         ................  [PASS] 
	Running test: testTornadoMathCosPI       ................  [PASS] 
	Running test: testTornadoMathCosPIDouble ................  [PASS] 
	Running test: testTornadoMathSinPI       ................  [PASS] 
	Running test: testTornadoMathSinPIDouble ................  [PASS] 
	Running test: testTornadoMathSignumFloat ................  [PASS] 
	Running test: testTornadoMathSin         ................  [PASS] 
	Running test: testTornadoMathAtan        ................  [PASS] 
	Running test: testTornadoMathAtanDouble  ................  [PASS] 
	Running test: testTornadoMathTan         ................  [PASS] 
	Running test: testTornadoMathTanDouble   ................  [PASS] 
	Running test: testTornadoMathTanh        ................  [PASS] 
	Running test: testTornadoMathTanhDouble  ................  [PASS] 
	Running test: testTornadoMathMin         ................  [PASS] 
	Running test: testTornadoMathMax         ................  [PASS] 
	Running test: testTornadoMathSqrt        ................  [PASS] 
	Running test: testTornadoMathExpDouble   ................  [PASS] 
	Running test: testTornadoMathExpFloat    ................  [PASS] 
	Running test: testTornadoMathClamp       ................  [PASS] 
	Running test: testTornadoMathFract       ................  [PASS] 
	Running test: testTornadoMathLog2        ................  [PASS] 
	Running test: testTornadoMathLogDouble   ................  [PASS] 
	Running test: testTornadoMathLogFloat    ................  [PASS] 
	Running test: testTornadoMathPI          ................  [PASS] 
	Running test: testMathFloor              ................  [PASS] 
	Running test: testMathClamp              ................  [PASS] 
	Running test: testTornadoMathRadians     ................  [PASS] 
	Running test: testTornadoMathRadiansDouble ................  [PASS] 
	Running test: testMathATan2              ................  [PASS] 
	Running test: testMathATan2Double        ................  [PASS] 
	Running test: testMathAcos               ................  [PASS] 
	Running test: testMathASin               ................  [PASS] 
```

**Review by mairooni — APPROVED** [review]
LGTM

File: tornado-unittests/src/main/java/uk/ac/manchester/tornado/unittests/math/TestTornadoMathCollection.java [hunk 1]
```diff
@@ -538,6 +568,36 @@ public void testTornadoMathAtan() throws TornadoExecutionPlanException {
...
+    @Test
+    public void testTornadoMathAtanDouble() throws TornadoExecutionPlanException {
+        assertNotBackend(TornadoVMBackendType.PTX);
+
+        final int size = 128;
+        DoubleArray data = new DoubleArray(size);
+        DoubleArray seq = new DoubleArray(size);
+
+        IntStream.range(0, size).parallel().forEach(i -> {
+            data.set(i, (float) Math.random());
```
Comment (stratika) [inline tornado-unittests/src/main/java/uk/ac/manchester/tornado/unittests/math/TestTornadoMathCollection.java:582]: I guess remove the casting, similar to the previous line (314).

**Review by stratika — APPROVED** [review]
Just a minor comment from my side.

**Pull request merged**